src/hooks/useInitialValues.ts

```diff
@@ -35,13 +35,20 @@ export function useInitialValues(
       Promise.all(asyncItemsKeys.map((key) => asyncInitValues[key])).then((results) => {
         asyncItemsKeys.forEach((key, index) => {
           const result = isEmpty(results[index]) ? '' : results[index];
-          initialValues[key] = result;
           const field = formFields.find((field) => field.id === key);
+          initialValues[key] = result;
           try {
             if (!isEmpty(result)) {
               formFieldHandlers[field.type].handleFieldSubmission(field, result, encounterContext);
             }
           } catch (error) {
+            const encounterValue = formFieldHandlers[field.type]?.getInitialValue(
+              encounter,
+              field,
+              formFields,
+              encounterContext,
+            );
+            formFieldHandlers[field.type].handleFieldSubmission(field, encounterValue, encounterContext);
```

Comment:
So we are only falling back to the encounter's value when we run into an error?

Reply:
yes